Multiple styles per layer

leendertvanwolfswinkel opened the pull request:
Structuur van de map ThreeDiToolbox\layer_styles\schematisation aangepast. Eerst stond daar bv. v2_connection_nodes.qml, nu staat daar een mapje 'v2_connection_nodes' met daarin een aantal styles. Daar moet altijd een default.qml staan, maar verder wordt alles wat er in staat toegevoegd aan de laag.

In utils\styler.py de functie apply_style aangepast. In plaats van 1 style te laden, laadt hij alle styles in de betreffende map in en stelt hem dan in op default.

In utils\layer_tree_manager.py heb ik de volgorde van de lagen v2_pumpstation en v2_pumpstation_point_view omgewisseld, omdat dat qua visualisatie duidelijker is

leendertvanwolfswinkel committed:
Multiple styles per layer

Structuur van de map ThreeDiToolbox\layer_styles\schematisation aangepast. Eerst stond daar bv. v2_connection_nodes.qml, nu staat daar een mapje 'v2_connection_nodes' met daarin een aantal styles. Daar moet altijd een default.qml staan, maar verder wordt alles wat er in staat toegevoegd aan de laag.

In utils\styler.py de functie apply_style aangepast. In plaats van 1 style te laden, laadt hij alle styles in de betreffende map in en stelt hem dan in op default.

In utils\layer_tree_manager.py heb ik de volgorde van de lagen v2_pumpstation en v2_pumpstation_point_view omgewisseld, omdat dat qua visualisatie duidelijker is

leendertvanwolfswinkel committed:
Styling finetuning

Most changes are minor improvements in the .qml styling files
Changes in layer_tree_manager: move the 1D layers that have a geometry but also a _view to the 'additional_oned_layers' group and toggle off their visibility, because 1) this more consistent (the 1D layers group contains all the views, because they give most information when inspecting the model, the 'additional 1d' group contains the layers where edits can be made) and 2) this made it easier to make a dynamic styling based on zoom level.

leendertvanwolfswinkel committed:
Repaired attribute forms in styling files for v2_pumpstation, v2_weir, v2_orifice

leendertvanwolfswinkel committed:
Minor improvements in styling for weir, culvert, orifice

Possible flow directions (based on discharge coef) were not visualized properly for v2_culvert_view. Put coalesce(x, 0) around all discharge_coefficient-related style rules.

RichardB9 commented on utils/styler.py:
```diff
@@ -10,9 +11,29 @@
-        layer.loadNamedStyle(style)
-    else:
-        logger.info("Style not exist {0}".format(style))
+    qml_path = os.path.join(STYLES_ROOT, stype, style_name)
```
Gebruik van de Pathlib maakt dit wat simpeler:

```python
from pathlib import Path

from qgis.core import QgsMapLayerStyle

logger = logging.getLogger(__name__)

STYLES_ROOT = Path(__file__).parent.parent / "layer_styles"


def apply_style(layer, style_name, stype="tools"):
    qml_path = STYLES_ROOT / stype / style_name

    style_manager = layer.styleManager()
    style_manager.reset()

    style = QgsMapLayerStyle()
    style.readFromLayer(layer)

    current_style_name = layer.styleManager().currentStyle()  # default style name depends on language settings

    for qml_file in qml_path.glob("*.qml"):
        style_name = qml_file.stem.capitalize()
        ...

leendertvanwolfswinkel replied:
Ha Richard,

Scherp van je. Ik ben de attribute forms van de andere styles ook nog even bij langs gelopen en heb nog wat meer (kleine) foutjes gevonden en gefixt. Heb ik gecommit op mijn fork.

Goeie van Pathlib, ga ik voortaan gebruiken!

RichardB9 reviewed (changes requested):
Mooie toevoeging al die extra styles!

Een paar kleine dingen die mij opvielen:
- In `v2_culvert_view.cross_section_definition_id` van `default.qml` is de attribute form veranderd van een "Text Edit" naar een "Relation Reference". Maar deze "relation reference" staat momenteel niet goed ingesteld. Waarschijnlijk is dit een foutje en moet dit een "Text Edit" blijven.
- Hetzelfde in de `v2_pipe.cross_section_definition_id` in de `calculation_type.qml` style.
- Gebruik van Pathlib, zie comment.

leendertvanwolfswinkel committed:
Minor changes to some styles

v2_culvert_view
	default - restored fields & forms
	diameter - restored fields & forms

v2_pipe
	calculation type - restored fields & forms

v2_manhole_view
	inspection manhole white instead of transparant for  default, code, levels

v2_pumpstation_view
	levels - restored fields & forms
